### PR TITLE
Lesson 9 Assignment files

### PR DESCRIPTION
**Pull request by zaandrew**
Lesson 9 Assignment files

### CODE REVIEW

File: students/andrey_zkn/lesson09/donor_models.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        for name in self.donors:
+            lod.append(name)
+        return '\n'.join(lod)
+      
+    def send_letters_to_all(self):
+        """ Save donor letters to files """
+        for donor in self.donors.values():
+            filename = donor.name + '.txt'
+            with open(filename, 'w') as f:
+                f.write(Donor.print_thank_you_message(donor))
```
Comment (natasha-aleksandrova) [inline students/andrey_zkn/lesson09/donor_models.py:61]: this is an incorrect usage of the class here, you should call the method on the instance:
```
donor.print_thank_you_message()
```

File: students/andrey_zkn/lesson09/donor_models.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        for entry in self.donors.values():
+            total = sum(entry.donation_amt)
+            num = len(entry.donation_amt)
+            avg = total/num
+            print('{:<20} ${:>14,.2f}{:>14}  ${:>16,.2f}'.format(entry.name,total,num,avg))
+        print('')
+  
+    def donor_exists(self, name):
+        #Search for donor name in donor collection.    
+        return any(d for d in self.donors.values() if d.name.upper() == name.upper())
```
Comment (natasha-aleksandrova) [inline students/andrey_zkn/lesson09/donor_models.py:79]: this is not necessary - you should use dictionary look up syntax by key

File: students/andrey_zkn/lesson09/donor_models.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                                                   "Num Gifts", "Average Gift")
+        print("Donation Report")
+        print(header)
+        print('-'*len(header))
+        for entry in self.donors.values():
+            total = sum(entry.donation_amt)
+            num = len(entry.donation_amt)
+            avg = total/num
+            print('{:<20} ${:>14,.2f}{:>14}  ${:>16,.2f}'.format(entry.name,total,num,avg))
+        print('')
```
Comment (natasha-aleksandrova) [inline students/andrey_zkn/lesson09/donor_models.py:75]: missing sorting here

**Comment by zaandrew** [comment]
Thank you for providing this feedback. It is really helpful.

**Pull request merged**